src/uu/df/src/df.rs

```diff
@@ -274,8 +298,11 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
 
     let opt = Options::from(&matches);
 
-    let mounts = read_fs_list();
-    let data: Vec<Row> = filter_mount_list(mounts, &paths, &opt)
+    let mut mounts = read_fs_list();
+
+    // Sort mounts in desc ordered lexicographically
+    mounts.sort_by(|a, b| b.mount_dir.cmp(&a.mount_dir));
```

Comment:
Why sort here? On my system (Ubuntu), the output of `df` is not lexicographically ordered:

```
$ df --output=source | tail -n +2 | head -n4
udev
tmpfs
/dev/mapper/vgubuntu-root
tmpfs
```

See also my comment here: https://github.com/uutils/coreutils/issues/3086#issuecomment-1030965538

Reply:
My plan was to sort in the order of decreasing path length, so that when I do a prefix matching, the longest matching mount path is selected first and rest paths are ignored. For example if we have mount paths [`\`,`\root`] and we have input `/root/docs` then we only need to output `/root` and `/` needs to be skipped. Sorting helps to assure this part

Let me update the sorting section and remove lexicographical part and change to sort by length.